sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/pipeline/TelemetryItemExporter.java

```diff
@@ -176,41 +150,6 @@ private TelemetryItem createOtelResourceMetric(TelemetryItemBatchKey telemetryIt
         return builder.build();
     }
 
-    List<ByteBuffer> encode(List<TelemetryItem> telemetryItems) throws IOException {
-        if (logger.canLogAtLevel(LogLevel.VERBOSE)) {
-            StringWriter debug = new StringWriter();
-            try (JsonGenerator jg = mapper.createGenerator(debug)) {
-                writeTelemetryItems(jg, telemetryItems);
-            }
-            logger.verbose("sending telemetry to ingestion service:{}{}", System.lineSeparator(), debug);
-        }
-
-        ByteBufferOutputStream out = new ByteBufferOutputStream(byteBufferPool);
-
-        try (JsonGenerator jg = mapper.createGenerator(new GZIPOutputStream(out))) {
-            writeTelemetryItems(jg, telemetryItems);
-        } catch (IOException e) {
-            byteBufferPool.offer(out.getByteBuffers());
-            throw e;
-        }
-
-        out.close(); // closing ByteBufferOutputStream is a no-op, but this line makes LGTM happy
-
-        List<ByteBuffer> byteBuffers = out.getByteBuffers();
-        for (ByteBuffer byteBuffer : byteBuffers) {
-            byteBuffer.flip();
-        }
-        return byteBuffers;
-    }
-
-    private static void writeTelemetryItems(JsonGenerator jg, List<TelemetryItem> telemetryItems)
-        throws IOException {
-        jg.setRootValueSeparator(new SerializedString("\n"));
-        for (TelemetryItem telemetryItem : telemetryItems) {
-            mapper.writeValue(jg, telemetryItem);
-        }
-    }
-
```

Comment:
why extract these to a new class?

Reply:
reusability

Reply:
maybe I missed, but which methods are being used by more than one (non-test) class?

Reply:
all the public methods are used more than once. feel free to check out the PR and verify yourself.

Reply:
ok, this is what I found:

* PipelineUtil.deserializeTelemetryItem is only used by LocalStorageTelemetryPipelineListener
* PipelineUtil.decode is only used by LocalStorageTelemetryPipelineListener

Reply:
and tests

and 
![image](https://github.com/Azure/azure-sdk-for-java/assets/56097766/cd27fa33-abc4-4125-b448-2cdcd6f32256)

Reply:
it's my preference.

Reply:
for methods that are only called from a single (non-Test) class, I'd recommend collocating them with the code that calls them, as it makes it easier for other people to read and follow

if the class was very big, that could be a good reason to violate this rule and split the code out, but that doesn't seem to be the case here

if you need to make them visible for tests, you can then mark them as [`// visible for testing`](https://github.com/search?q=repo%3Aopen-telemetry%2Fopentelemetry-java-instrumentation%20visible%20for%20testing&type=code)